src/rgw/rgw_asio_frontend.cc

```diff
@@ -1011,6 +1011,10 @@ void AsioFrontend::accept(Listener& l, boost::system::error_code ec)
     return;
   } else if (ec) {
     ldout(ctx(), 1) << "accept failed: " << ec.message() << dendl;
+    l.acceptor.async_accept(l.socket,
+                            [this, &l] (boost::system::error_code ec) {
+                              accept(l, ec);
+                            });
```

Comment:
this does seem like the right thing to do for temporary errors like file descriptor limits 👍 

but a lot of the error codes listed in https://www.man7.org/linux/man-pages/man2/accept.2.html#ERRORS are not temporary errors. if we enter a retry loop on any of those, we'll just spin here forever and fill the rgw log with "accept failed" messages

even file descriptor errors may take a while to clear up (seconds to minutes?). spinning here could hog the cpu and prevent other requests from completing/releasing their fds

maybe we should add some backoff to these retries?

Reply:
i opened a draft pr https://github.com/ceph/ceph/pull/59059 that does this. if you think that's useful, feel free to add the backoff/retry logic on top

Reply:
> i opened a draft pr #59059 that does this. if you think that's useful, feel free to add the backoff/retry logic on top

yep, it's very useful and you already writed a loop to call async_accept. It is easy to backoff/retry based the work. I have learned a lot from this, and you can also add retry codes directly if you make the pr as draft just because i commit origin one.